packages/boot/test/unit/utils.unit.ts

```diff
@@ -0,0 +1,194 @@
+      expect(files).to.be.a.Array();
+      expect(files).to.have.lengthOf(0);
+    });
+  });
+
+  describe('flatten()', () => {
+    it('flattens an array of arrays', () => {
+      const input = ['a', ['b', 'c'], 'd', ['e', ['f']]];
+      const expected: string[] = ['a', 'b', 'c', 'd', 'e', 'f'];
+      const actual: string[] = flatten(input);
+      expect(actual.sort()).to.be.eql(expected.sort());
+    });
+  });
+
+  describe('filterExts()', () => {
+    const input = ['a.ts', 'b.ts', 'a.js', 'b.js', 'a.js.map', '.b.js.map'];
+    it('filters an array for matching extension', () => {
+      const filter: string[] = ['.ts'];
+      const expected: string[] = ['a.ts', 'b.ts'];
+      const result: string[] = filterExts(input, filter);
+      expect(result.sort()).to.be.eql(expected.sort());
+    });
+
+    it('filters an array for matching extensions', () => {
+      const filter: string[] = ['.ts', '.js'];
+      const expected: string[] = ['a.ts', 'b.ts', 'a.js', 'b.js'];
+      const result: string[] = filterExts(input, filter);
+      expect(result.sort()).to.be.eql(expected.sort());
+    });
+
+    it('returns input when no filter is applied', () => {
+      const result: string[] = filterExts(input, []);
+      expect(result.sort()).to.be.eql(input.sort());
+    });
+  });
+
+  describe('normalizeToArray()', () => {
+    it('converts a string to an array', () => {
+      const result: string[] = normalizeToArray('hello');
+      expect(result).to.be.an.Array();
+      expect(result).to.be.eql(['hello']);
+    });
+
+    it('returns an array given an array', () => {
+      const input: string[] = ['hello'];
+      const result: string[] = normalizeToArray(input);
+      expect(result).to.be.eql(input);
+    });
+  });
+
+  describe('discoverArtifactFiles()', () => {
+    before(getRootDir);
+
+    it('discovers correct artifact files in nested folders with Array input', () => {
+      const dirs: string[] = [resolve(rootDir)];
+      const exts: string[] = ['.controller.js'];
+      const expected: string[] = [
+        resolve(rootDir, 'controllers/hello.controller.js'),
+        resolve(rootDir, 'controllers/admin/admin.controller.js'),
+      ];
+      const result: string[] = discoverArtifactFiles(dirs, exts);
+      expect(result.sort()).to.be.eql(expected.sort());
+    });
+
+    it('discovers correct artifact files in nested folders with String input', () => {
+      const expected: string[] = [
+        resolve(rootDir, 'controllers/hello.controller.js'),
+        resolve(rootDir, 'controllers/admin/admin.controller.js'),
+      ];
+      const result: string[] = discoverArtifactFiles(
+        resolve(rootDir),
+        '.controller.js',
+      );
+      expect(result.sort()).to.be.eql(expected.sort());
+    });
+
+    it('discovers correct artifact files and ignores nested folders', () => {
+      const dirs: string[] = [resolve(rootDir)];
+      const exts: string[] = ['.controller.js'];
```

Comment:
Please import these values from the constants you exported above (see previous comment).